lib/shared/bucketing/src/segmentation.ts

```diff
@@ -112,39 +154,97 @@ function filterForAudienceMatch({
-    return comparator === '!='
+    return {
+        result: comparator === '!=',
+        reasonDetails: 'Not in Audience',
+    }
 }
 
 const filterFunctionsBySubtype: FilterFunctionsBySubtype = {
-    country: (data, filter) => checkStringsFilter(data.country, filter),
-    email: (data, filter) => checkStringsFilter(data.email, filter),
-    ip: (data, filter) => checkStringsFilter(data.ip, filter),
-    user_id: (data, filter) => checkStringsFilter(data.user_id, filter),
-    appVersion: (data, filter) => checkVersionFilters(data.appVersion, filter),
-    platformVersion: (data, filter) =>
-        checkVersionFilters(data.platformVersion, filter),
-    deviceModel: (data, filter) => checkStringsFilter(data.deviceModel, filter),
-    platform: (data, filter) => checkStringsFilter(data.platform, filter),
+    country: (data, filter) => {
+        const result = checkStringsFilter(data.country, filter)
+        return {
+            result,
+            reasonDetails: result ? `Country` : undefined,
+        }
+    },
+    email: (data, filter) => {
+        const result = checkStringsFilter(data.email, filter)
+        return {
+            result,
+            reasonDetails: result ? `Email` : undefined,
+        }
+    },
+    ip: (data, filter) => {
+        const result = checkStringsFilter(data.ip, filter)
+        return {
+            result,
+            reasonDetails: result ? 'IP' : undefined,
+        }
+    },
+    user_id: (data, filter) => {
+        const result = checkStringsFilter(data.user_id, filter)
+        return {
+            result,
+            reasonDetails: result ? 'User ID' : undefined,
+        }
+    },
+    appVersion: (data, filter) => {
+        const result = checkVersionFilters(data.appVersion, filter)
+        return {
+            result,
+            reasonDetails: result ? 'App Version' : undefined,
+        }
+    },
+    platformVersion: (data, filter) => {
+        const result = checkVersionFilters(data.platformVersion, filter)
+        return {
+            result,
+            reasonDetails: result ? 'Platform Version' : undefined,
+        }
+    },
+    deviceModel: (data, filter) => {
+        const result = checkStringsFilter(data.deviceModel, filter)
+        return {
+            result,
+            reasonDetails: result ? 'Device Model' : undefined,
+        }
+    },
+    platform: (data, filter) => {
+        const result = checkStringsFilter(data.platform, filter)
+        return {
+            result,
+            reasonDetails: result ? 'Platform' : undefined,
+        }
+    },
     customData: (data, filter) => {
         const combinedCustomData = {
             ...data.customData,
             ...data.privateCustomData,
         }
-        return checkCustomData(combinedCustomData, filter)
+        const result = checkCustomData(combinedCustomData, filter)
+        return {
+            result,
+            reasonDetails: result ? `Custom Data ${filter.dataKey}` : undefined,
```

Comment:
i would make this also generically apply to say "custom data or private custom data"

Reply:
From a filter perspective, its just a Custom Data match